ctru-sys/bindgen-ctru-sys/src/main.rs

```diff
@@ -0,0 +1,61 @@
+use bindgen::callbacks::ParseCallbacks;
+use bindgen::{Builder, RustTarget};
+use std::path::PathBuf;
+
+#[derive(Debug)]
+struct CustomCallbacks;
+
+impl ParseCallbacks for CustomCallbacks {
+    fn process_comment(&self, comment: &str) -> Option<String> {
+        Some(doxygen_rs::transform(comment))
+    }
+}
+
+fn main() {
+    let devkitpro = std::env::var("DEVKITPRO").expect("DEVKITPRO not set in environment");
+    let devkitarm = std::env::var("DEVKITARM").expect("DEVKITARM not set in environment");
+
+    let include_path = PathBuf::from_iter([devkitpro.as_str(), "libctru", "include"]);
+    let header = include_path.join("3ds.h");
+
+    let sysroot = PathBuf::from(devkitarm).join("arm-none-eabi");
+    let system_include = sysroot.join("include");
+
+    let bindings = Builder::default()
+        .header(header.to_str().unwrap())
+        .rust_target(RustTarget::Nightly)
+        .use_core()
+        .trust_clang_mangling(false)
+        .must_use_type("Result")
+        .layout_tests(false)
+        .ctypes_prefix("::libc")
+        .prepend_enum_name(false)
+        .blocklist_type("u(8|16|32|64)")
+        .blocklist_type("__builtin_va_list")
+        .blocklist_type("__va_list")
+        .opaque_type("MiiData")
+        .derive_default(true)
+        .clang_args([
+            "--target=arm-none-eabi",
+            "--sysroot",
+            sysroot.to_str().unwrap(),
+            "-isystem",
+            system_include.to_str().unwrap(),
+            "-I",
+            include_path.to_str().unwrap(),
+            "-mfloat-abi=hard",
+            "-march=armv6k",
+            "-mtune=mpcore",
+            "-mfpu=vfp",
+            "-DARM11 ",
```

Comment:
are the trailing spaces here intentional? I'm not sure how clang behaves here.

Also, I think as of [libctru 2.1.0](https://github.com/devkitPro/libctru/releases/tag/v2.1.0) only `__3DS__` should be needed, not `_3DS`. But probably no harm in having both

Reply:
I took a look at the code from citro-3d-bindgen, the shell script and the main.rs had the same params so I just put the same 😄

So, yeah, I just took it from citro-3d

Reply:
> I think as of [libctru 2.1.0](https://github.com/devkitPro/libctru/releases/tag/v2.1.0) only `__3DS__` should be needed, not `_3DS`. But probably no harm in having both

We should only ask for `__3DS__`, in my opinion. Libraries which haven't updated to the latest standard also won't be updated for the latest `libctru`, so we won't really be losing any compatibility. Let's just stick to the standard.

Also, the trailing spaces should be removed, I don't think they break anything, but it's not pretty.